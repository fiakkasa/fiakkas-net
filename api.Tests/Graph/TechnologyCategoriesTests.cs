using HotChocolate.Execution;

namespace api.Tests;

public class TechnologyCategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task TechnologyCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  technologyCategories {
    totalCount
    nodes {
      createdAt
      href
      id
      internalId
      kind
      title
      updatedAt
      version
      portfolioCategories {
        totalCount
      }
      portfolioCustomers {
        totalCount
      }
      portfolioItems {
        totalCount
      }
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task TechnologyCategoryById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  technologyCategoryById(id: "VGVjaG5vbG9neUNhdGVnb3J5CmdjYTgzMmJmOWI3Y2I0YzMxYmY4ZDAwZjg3YTI3NmZlMw==") {
    createdAt
    href
    id
    internalId
    kind
    title
    updatedAt
    version
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
