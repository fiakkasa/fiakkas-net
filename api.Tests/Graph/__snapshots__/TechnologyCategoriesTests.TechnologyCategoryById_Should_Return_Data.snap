{
  "data": {
    "technologyCategoryById": {
      "createdAt": "2024-01-01T00:00:00.000Z",
      "href": "/test",
      "id": "VGVjaG5vbG9neUNhdGVnb3J5CmdjYTgzMmJmOWI3Y2I0YzMxYmY4ZDAwZjg3YTI3NmZlMw==",
      "internalId": "ca832bf9-b7cb-4c31-bf8d-00f87a276fe3",
      "kind": "SOFTWARE_DEVELOPMENT",
      "title": "Title",
      "updatedAt": null,
      "version": 1
    }
  }
}
